envoy/ssl/connection.h

```diff
@@ -95,6 +96,11 @@ class ConnectionInfo {
    **/
   virtual const std::string& subjectPeerCertificate() const PURE;
 
+  /**
+   * @return the well-known attribute values parsed from subject field of the peer certificate.
+   **/
+  virtual const ParsedX509NameConstSharedPtr& parsedSubjectPeerCertificate() const PURE;
```

Comment:
@ggreenway just want to confirm it's still ok to use shared_ptr in the member right (so we don't allocate memory until someone called the function)?

Reply:
Use a `unique_ptr` instead

Reply:
@ggreenway  if return reference then we can't use null pointer and caller can't check if the subject not exist, such as the change in quic_ssl_connection_info.h which must allocate the object. so looks OptRef is the only choice. 

That being said, my understanding is that the OptRef is just a wrapper of raw pointer, which is less safe than shared_ptr because it can still keep the pointer when the original object is deleted and will be used to reference an invalid address. shared_ptr will make sure the object is kept, although it has risk of memory leak but still better than invalid memory access I guess.